artifactory/services/upload.go

```diff
@@ -765,6 +765,13 @@ func (us *UploadService) addFileToZip(artifact *clientutils.Artifact, progressPr
 		header.Name = clientutils.TrimPath(localPath)
 	}
 	header.Method = zip.Deflate
+
+	// If this is a directory, add it to the writer with a trailing slash.
+	if info.IsDir() {
+		header.Name = header.Name + "/"
```

Comment:
Can be shortened to - 
```
header.Name += "/"
```
Also, do we always need to add the slash? Can the name sometimes end with a slash?